src/targets/gpu/fuse_mlir.cpp

```diff
@@ -44,22 +44,20 @@ struct mlir_op
 {
     std::string name() const { return "gpu::mlir_op"; }
     operation op = make_op("convolution");
+    shape output_shape;
 
     template <class Self, class F>
     static auto reflect(Self& self, F f)
     {
-        return pack(f(self.op, "op"));
+        return pack(f(self.op, "op"), f(self.output_shape, "output_shape"));
     }
 
     shape compute_shape(std::vector<shape> inputs, const std::vector<module_ref>& mods) const
     {
-        check_shapes{inputs, *this}.packed_or_broadcasted();
```

Comment:
This should not be removed as MLIR doesnt support non-packed tensors as inputs.

Reply:
restored.